src/sql/parts/notebook/notebookActions.ts

```diff
@@ -336,14 +336,14 @@ export class AttachToDropdown extends SelectBox {
 			await this._connectionDialogService.openDialogAndWait(this._connectionManagementService, { connectionType: 1, providers: this.model.getApplicableConnectionProviderIds(this.model.clientSession.kernel.name) }).then(connection => {
 				let attachToConnections = this.values;
 				if (!connection) {
-					this.loadAttachToDropdown(this.model, this.model.clientSession.kernel.name);
+					this.loadAttachToDropdown(this.model, this.getKernelDisplayName());
```

Comment:
So what was the problem here? what was the value for this.model.clientSession.kernel.name versus what getKernelDisplayName() returns?

Reply:
It is 'pyspark3kernel' where as display name is PySpark3. _kernelDisplayNameToConnectionProviderIds gets you list of providers based on kernel display name. Since there is now provider id Attach To is being set to Localhost

Reply:
ah, gotcha